pkg/network/ovn_kubernetes.go

```diff
@@ -1738,3 +2214,283 @@ func isV6InternalSubnetLargeEnough(conf *operv1.NetworkSpec) bool {
+	return isMigrationToMultiZoneAboutToStart(ovn, targetZoneMode, false) || isMigrationToMultiZoneOngoing(ovn, targetZoneMode)
+}
+
+// migration from multizone to single zone is about to start if target is single zone and the cluster
+// is on multizone:
+// - node is running, is >= 4.14, is in multizone and is not progressing
+// - master is not running
+// - control plane is running and is not progressing
+func isMigrationToSingleZoneAboutToStart(ovn bootstrap.OVNBootstrapResult, targetZoneMode *targetZoneModeType) bool {
+	return targetZoneMode.zoneMode == zoneModeSingleZone &&
+
+		ovn.NodeUpdateStatus != nil &&
+		ovn.MasterUpdateStatus == nil &&
+		ovn.ControlPlaneUpdateStatus != nil &&
+
+		ovn.NodeUpdateStatus.InterConnectEnabled &&
+		ovn.NodeUpdateStatus.InterConnectZoneMode == string(zoneModeMultiZone) &&
+
+		!ovn.NodeUpdateStatus.Progressing &&
+		!ovn.ControlPlaneUpdateStatus.Progressing
+}
+
+// migration from multizone to single zone first rolls out master, then node. It is ongoing if:
+// - target is single zone
+// - node is running, is >=4.14
+// - master is running, is >= 4.14, it rolls out when node is still in multizone;
+// - control plane is not running when node is rolling out in single zone
+func isMigrationToSingleZoneOngoing(ovn bootstrap.OVNBootstrapResult, targetZoneMode *targetZoneModeType) bool {
+	return targetZoneMode.zoneMode == zoneModeSingleZone &&
+
+		ovn.NodeUpdateStatus != nil && ovn.NodeUpdateStatus.InterConnectEnabled &&
+
+		ovn.MasterUpdateStatus != nil &&
+		ovn.MasterUpdateStatus.InterConnectEnabled &&
+
+		// node is not progressing (and in multizone) && master is progressing && master is single zone
+		(!ovn.NodeUpdateStatus.Progressing &&
+			ovn.NodeUpdateStatus.InterConnectZoneMode == string(zoneModeMultiZone) &&
+			ovn.MasterUpdateStatus.Progressing &&
+			ovn.MasterUpdateStatus.InterConnectZoneMode == string(zoneModeSingleZone) ||
+
+			// OR node is progressing and in single zone && master is not progressing && control plane was removed
+			ovn.NodeUpdateStatus.Progressing &&
+				ovn.NodeUpdateStatus.InterConnectZoneMode == string(zoneModeSingleZone) &&
+				!ovn.MasterUpdateStatus.Progressing &&
+				ovn.ControlPlaneUpdateStatus == nil)
+}
+
+func isMigrationToSingleZoneAboutToStartOrOngoing(ovn bootstrap.OVNBootstrapResult, targetZoneMode *targetZoneModeType) bool {
+	return isMigrationToSingleZoneAboutToStart(ovn, targetZoneMode) || isMigrationToSingleZoneOngoing(ovn, targetZoneMode)
+}
+
+func isZoneModeMigrationAboutToStartOrOngoing(ovn bootstrap.OVNBootstrapResult, targetZoneMode *targetZoneModeType) bool {
+	return isMigrationToMultiZoneAboutToStartOrOngoing(ovn, targetZoneMode) || isMigrationToSingleZoneAboutToStartOrOngoing(ovn, targetZoneMode)
+}
+
+func doesVersionEnableInterConnect(string) bool {
+	return isVersionGreaterThanOrEqualTo(os.Getenv("RELEASE_VERSION"), 4, 14)
```

Comment:
I added a new function because I just needed to compare the major and minor bits of a version and disregard all the rest. 
The function above should read: everything >= 4.14 should have IC on. If look at semver, it's telling me that the order is:

`Example: 1.0.0-alpha < 1.0.0-alpha.1 < 1.0.0-alpha.beta < 1.0.0-beta < 1.0.0-beta.2 < 1.0.0-beta.11 < 1.0.0-rc.1 < 1.0.0.`
(point 11 https://semver.org/#spec-item-11)

So `14.0.0-rc.1 < 14.0.0` and bye bye IC :-D  I think it's safer to use the simple function I came up with, even though I would have been happier to use an existing one :)